### PR TITLE
Niganesh/support for ed25519 curve

### PR DESCRIPTION
**Pull request by nithyaganeshng**
Supports ED25519 signature verification. When request is signed using ED25519 keys, it can now be verified.

**Validation:**
Presentation request signed using ED25519 keys were successfully verified.

**Type of change:**
- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: sdk/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jws/JwsToken.kt [hunk 1]
```diff
@@ -49,6 +51,16 @@ class JwsToken constructor(
         return false
     }
 
+    fun verifyUsingOctetKeyPair(publicKeys: List<OctetKeyPair> = emptyList()): Boolean {
```
Comment (symorton) [inline sdk/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jws/JwsToken.kt:54]: just curious: why is it called Octet Key Pair?

File: sdk/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jws/JwsToken.kt [hunk 2]
```diff
@@ -49,6 +51,16 @@ class JwsToken constructor(
...
+        for (key in publicKeys) {
+            val verifier = Ed25519Verifier(key)
+            if (jwsObject.verify(verifier)) {
```
Comment (symorton) [inline sdk/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jws/JwsToken.kt:57]: should we just be verifying with the public key with the matching keyId from the header of the token?

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/validators/JwtValidator.kt [hunk 3]
```diff
@@ -29,7 +32,14 @@ class JwtValidator @Inject constructor(
...
         val publicKeys = resolvePublicKey(didInHeader, keyIdInHeader)
-        return token.verify(publicKeys)
+        return verifySignatureUsingPublicKey(token, publicKeys)
+    }
+
+    private fun verifySignatureUsingPublicKey(token: JwsToken, publicKeys: List<JWK>): Boolean {
+        return if (publicKeys[0].keyType.value == ED25519_CURVE_KTY)
+            token.verifyUsingOctetKeyPair(listOf(publicKeys.first().toOctetKeyPair()))
+        else
+            token.verify(listOf(KeyConverter.toJavaKeys(publicKeys).first() as PublicKey))
```
Comment (AndiMiko) [inline sdk/src/main/java/com/microsoft/did/sdk/credential/service/validators/JwtValidator.kt:42]: I'd recommend moving this logic into `JwsToken.verify()`

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/validators/JwtValidator.kt [hunk 4]
```diff
@@ -29,7 +32,14 @@ class JwtValidator @Inject constructor(
         val (didInHeader: String?, keyIdInHeader: String) = getDidAndKeyIdFromHeader(token)
         if (didInHeader == null) throw ValidatorException("JWS contains no DID")
         val publicKeys = resolvePublicKey(didInHeader, keyIdInHeader)
```
Comment (AndiMiko) [inline sdk/src/main/java/com/microsoft/did/sdk/credential/service/validators/JwtValidator.kt:34]: nit, naming (not a public key anymore)

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/validators/JwtValidator.kt [hunk 5]
```diff
@@ -42,12 +52,12 @@ class JwtValidator @Inject constructor(
         throw ValidatorException("JWS contains no key id")
     }
 
-    private suspend fun resolvePublicKey(did: String, keyId: String): List<PublicKey> {
+    private suspend fun resolvePublicKey(did: String, keyId: String): List<JWK> {
```
Comment (AndiMiko) [inline sdk/src/main/java/com/microsoft/did/sdk/credential/service/validators/JwtValidator.kt:55]: nit, naming

**Review by AndiMiko — APPROVED** [review]

File: sdk/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jws/JwsToken.kt (hunk 1 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jws/JwsToken.kt:54]: That is the key type used to represent keys for ED curve.

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/validators/JwtValidator.kt (hunk 5 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/credential/service/validators/JwtValidator.kt]: done

File: sdk/src/main/java/com/microsoft/did/sdk/credential/service/validators/JwtValidator.kt (hunk 3 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/credential/service/validators/JwtValidator.kt]: done

File: sdk/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jws/JwsToken.kt (hunk 2 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/crypto/protocols/jose/jws/JwsToken.kt]: What would be passed here from validator is a list with just a public key that matches the keyId in the token header. But we support passing a list of public keys in general.

**Pull request merged**